src/AlgorithmiaClient.ts

```diff
@@ -226,13 +229,42 @@ class AlgorithmiaClient {
    * @param requestObject payload
    * @return an empty response
    */
-  editOrganization(orgName: string, requestObject: Input) {
+  async editOrganization(orgName: string, requestObject: Input) {
     return this.httpClient.put(
-      this.apiAddress + this.organizationPrefix + '/' + orgName,
-      JSON.stringify(requestObject)
+      `${this.apiAddress}${this.organizationsPrefix}/${orgName}`,
+      JSON.stringify(await this.organizationTypeIdChanger(requestObject))
     );
   }
 
+  /**
+   * Helper for swapping out the type_id value
+   */
+  async organizationTypeIdChanger(requestObject: Input) {
+    let editedOrganization: Organization = JSON.parse(JSON.stringify(requestObject));
+    let isSet = false;
+    if (!this.typesMapList.length) {
+      this.typesMapList = await this.getOrgTypes();
+    }
+    for (var type of this.typesMapList) {
+      if(type.name === editedOrganization.type_id) {
+        editedOrganization.type_id = type.id;
+        isSet = true;
+        break;
+      };
+    }
+    if (!isSet) {
+      throw new TypeError("No matching organization type found, should be one of 'legacy', 'basic', 'pro'");
```

Comment:
`TypeError` in JavaScript is used in cases where something can't happen because a JavaScript value is of the unexpected type (e.g., string vs null). In this instance the expected value and the actual value are both still objects, so a TypeError in that context is slightly misleading. I like your thinking on this, but it may confuse JS devs.

```suggestion
      throw new Error("No matching organization type found, should be one of 'legacy', 'basic', 'pro'");
```